provider/gateway/client.go

```diff
@@ -194,7 +221,19 @@ func (c *client) ServiceLogs(ctx context.Context,
 
 	endpoint.RawQuery = query.Encode()
 
-	conn, _, err := websocket.DefaultDialer.DialContext(ctx, endpoint.String(), nil)
+	conn, response, err := websocket.DefaultDialer.DialContext(ctx, endpoint.String(), nil)
+	if err == websocket.ErrBadHandshake {
+		buf := &bytes.Buffer{}
+		_, err = io.Copy(buf, response.Body)
+		if err != nil {
+			return nil, err
+		}
+		return nil, ClientResponseError{
+			Status:  response.StatusCode,
+			Message: buf.String(),
+		}
```

Comment:
perhaps encapsulate this pattern in a utility function?

Reply:
This has been done